### PR TITLE
vultr: fix for unreliable API behavior

### PR DESCRIPTION
**Pull request by resmo**
##### SUMMARY
currently we run to often into a timeout because we wait for state server_state=ok which has been increased dramatically since the development of the module. It can takes 5 or more minutes to get into this state, even when the VM is fully booted.

Another issue is that we should use retry also for all return codes other then HTTP 200, not only for API rate limits or server errors. This increases reliability.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

vultr_server

##### ANSIBLE VERSION

```paste below
2.6
```

##### ADDITIONAL INFORMATION

```paste below
should be backported to 2.6, 2.7
```

### CODE REVIEW

File: lib/ansible/module_utils/vultr.py [hunk 1]
```diff
@@ -169,8 +169,7 @@ def api_query(self, path="/", method="GET", data=None):
                 timeout=self.api_config['api_timeout'],
             )
 
-            # Did we hit the rate limit?
-            if info.get('status') and info.get('status') != 503:
+            if info.get('status') and info.get('status') == 200:
```
Comment (Spredzy) [inline lib/ansible/module_utils/vultr.py]: In general this can be simpliefied to `if info.get('status) == 200`

File: lib/ansible/modules/cloud/vultr/vultr_server.py [hunk 2]
```diff
@@ -720,7 +716,7 @@ def absent_server(self):
                     method="POST",
                     data=data
                 )
-                for s in range(0, 30):
+                for s in range(0, 60):
```
Comment (Spredzy) [inline lib/ansible/modules/cloud/vultr/vultr_server.py:719]: nit question: Why this change? Is it taking more time to get a server deleted? Also, should this be a parameter, (maybe reuse `api_timeout`) ? Is the behavior observed for `server_state` update also observed here or is it to play safe?

File: lib/ansible/modules/cloud/vultr/vultr_server.py [hunk 3]
```diff
@@ -764,7 +760,7 @@ def reinstall_server(self):
     def _wait_for_state(self, key='power_status', state=None):
         time.sleep(1)
         server = self.get_server(refresh=True)
-        for s in range(0, 30):
+        for s in range(0, 60):
```
Comment (Spredzy) [inline lib/ansible/modules/cloud/vultr/vultr_server.py:763]: ditto

**Review by Spredzy — APPROVED** [review]
Patch LGTM, minor remarks that can be addressed in a later patchset if needed.

File: lib/ansible/module_utils/vultr.py (hunk 1 above)
Comment (resmo) [reply lib/ansible/module_utils/vultr.py]: you are right!

File: lib/ansible/modules/cloud/vultr/vultr_server.py (hunk 2 above)
Comment (resmo) [reply lib/ansible/modules/cloud/vultr/vultr_server.py:719]: I will make it configurable but first, I would like to get this in as a bug fix for 2.7. the timeout was 30 times 2 seconds, I just wanted to make it "engough" long for 2.7 (it seems the time to wait has dramatically increased)

Comment (Spredzy) [reply lib/ansible/modules/cloud/vultr/vultr_server.py:719]: Ack, shipit'able for me. Do you want to change the `info.get('status')` first?
